server/src-lib/Hasura/Server/Logging.hs

```diff
@@ -118,118 +117,133 @@ instance ToJSON WebHookLog where
+instance ToJSON HttpInfoLog where
+  toJSON (HttpInfoLog st met src path hv) =
     object [ "status" .= N.statusCode st
            , "method" .= met
            , "ip" .= src
            , "url" .= path
            , "http_version" .= show hv
-           , "detail" .= det
-           , "request_id" .= reqId
-           , "user" .= hUser
-           , "query_hash" .= qh
-           , "response_size" .= rs
-           , "query_execution_time" .= rt
            ]
 
-data LogDetail
-  = LogDetail
-  { _ldQuery :: !TBS.TByteString
-  , _ldError :: !Value
+-- | Information about a GraphQL operation over HTTP
+data OperationLog
+  = OperationLog
+  { olRequestId          :: !RequestId
+  , olUserVars           :: !(Maybe UserVars)
+  , olResponseSize       :: !(Maybe Int64)
+  , olQueryExecutionTime :: !(Maybe Double)
+  , olQuery              :: !(Maybe Value)
+  , olError              :: !(Maybe Value)
   } deriving (Show, Eq)
+$(deriveToJSON (aesonDrop 2 snakeCase) ''OperationLog)
 
-instance ToJSON LogDetail where
-  toJSON (LogDetail q e) =
-    object [ "request"  .= q
-           , "error" .= e
-           ]
+data HttpAccessLog
+  = HttpAccessLog
+  { halHttpInfo  :: !HttpInfoLog
+  , halOperation :: !OperationLog
+  } deriving (Show, Eq)
+$(deriveToJSON (aesonDrop 3 snakeCase) ''HttpAccessLog)
 
-ravenLogGen
-  :: (BL.ByteString, Either QErr BL.ByteString)
-  -> (N.Status, Maybe Value, Maybe T.Text, Maybe Int64)
-ravenLogGen (reqBody, res) =
-  (status, toJSON <$> logDetail, Just qh, Just size)
-  where
-    status = either qeStatus (const N.status200) res
-    logDetail = either (Just . qErrToLogDetail) (const Nothing) res
-    reqBodyTxt = TBS.fromLBS reqBody
-    qErrToLogDetail qErr =
-      LogDetail reqBodyTxt $ toJSON qErr
-    size = BL.length $ either encode id res
-    qh = T.pack . show $ sha1 reqBody
-    sha1 :: BL.ByteString -> Digest SHA1
-    sha1 = hash . BL.toStrict
+data HttpLog
+  = HttpLog
+  { _hlLogLevel :: !L.LogLevel
+  , _hlLogLing  :: !HttpAccessLog
+  }
+
+instance L.ToEngineLog HttpLog where
+  toEngineLog (HttpLog logLevel accessLog) =
+    (logLevel, ELTHttpLog, toJSON accessLog)
 
 mkAccessLog
   :: Maybe UserInfo -- may not have been resolved
-  -> Request
-  -> (BL.ByteString, Either QErr BL.ByteString)
+  -> RequestId
+  -> Wai.Request
+  -> BL.ByteString
   -> Maybe (UTCTime, UTCTime)
-  -> AccessLog
-mkAccessLog userInfoM req r mTimeT =
-  AccessLog
-  { alStatus       = status
-  , alMethod       = bsToTxt $ requestMethod req
-  , alSource       = bsToTxt $ getSourceFromFallback req
-  , alPath         = bsToTxt $ rawPathInfo req
-  , alHttpVersion  = httpVersion req
-  , alDetail       = mDetail
-  , alRequestId    = bsToTxt <$> getRequestId req
-  , alHasuraUser   = userVars <$> userInfoM
-  , alResponseSize = size
-  , alResponseTime = realToFrac <$> diffTime
-  , alQueryHash    = queryHash
-  }
+  -> HttpLog
+mkAccessLog userInfoM reqId req res mTimeT =
```

Comment:
`mkHttpLog`?

Reply:
`mkAccessLog` and `mkErrorLog` both gives `HttpLog`. One is on `success` and the other is on `error`. I think it's better to define functions accordingly
Like:-
- `mkSuccessHttpLog`
- `mkErrorHttpLog`